packages/core/src/getStateFromPath.tsx

```diff
@@ -0,0 +1,50 @@
+import { InitialState } from './types';
+
+/**
+ * Utility to parse a path string to initial state object accepted by the container.
+ * This is useful for deep linking when we need to handle the incoming URL.
+ *
+ * @param path Path string to parse and convert, e.g. /foo/bar?count=42.
+ */
+export default function getStateFromPath(path: string): InitialState {
+  const parts = path.split('?');
+  const segments = parts[0].split('/').filter(Boolean);
+  const query = parts[1] ? parts[1].split('&') : undefined;
+
+  let result: InitialState | undefined;
+  let current: InitialState | undefined;
+
+  while (segments.length) {
```

Comment:
Can't we make it with reduce?

Reply:
We can, but I feel it's simpler with a loop.